docker-compose.yml

```diff
@@ -1,21 +1,49 @@
-version: "2.2"
+version: "2.3"
 services:
+
   zookeeper:
     image: confluentinc/cp-zookeeper:5.4.1
     restart: always
     hostname: zookeeper
     container_name: zookeeper
+    ports:
+      - 2181:2181
     environment:
       ZOOKEEPER_SERVER_ID: 1
-      ZOOKEEPER_CLIENT_PORT: "2181"
-      ZOOKEEPER_TICK_TIME: "2000"
+      ZOOKEEPER_CLIENT_PORT: 2181
+      ZOOKEEPER_TICK_TIME: 2000
       KAFKA_OPTS: -Djava.security.auth.login.config=/etc/kafka/secrets/zookeeper_jaas.conf
                   -Dzookeeper.authProvider.1=org.apache.zookeeper.server.auth.SASLAuthenticationProvider
                   -DrequireClientAuthScheme=sasl
     volumes:
       - ./scripts/security:/etc/kafka/secrets
+
+  tools:
+    image: cnfltraining/training-tools:5.4
+    hostname: tools
+    container_name: tools
+    depends_on:
+      - zookeeper
+      - kafka1
+      - kafka2
+    volumes:
+      - ./scripts/helper:/tmp/helper
+    entrypoint: /bin/bash
+    tty: true
+
+  openldap:
+    image: osixia/openldap:1.3.0
+    hostname: openldap
+    container_name: openldap
     ports:
-      - "2181:2181"
+        - 389:389
+    environment:
+        LDAP_ORGANISATION: "Confluent"
+        LDAP_DOMAIN: "confluent.io"
+        LDAP_BASE_DN: "dc=confluent,dc=io"
```

Comment:
Not suggesting this is wrong, but wondering if it would be better to use a fake org name here, or a modification of Confluent e.g. `confluentdemo`, to save any possible confusion.

Also we are using the container-default admin password - a test/demo system so unlikely to be an issue - but maybe the LDAP instance is accessible remotely and someone accidentally stores a valuable credential?  Trade-off I guess between extra security diligence and convenience.  If generating a password it could be relayed to container stdout.

Accepted that this is a demo, non-production config so take with a grain of salt.

Reply:
> Not suggesting this is wrong, but wondering if it would be better to use a fake org name here, or a modification of Confluent e.g. confluentdemo, to save any possible confusion.

Great suggestion, changed to `confluentdemo`